### PR TITLE
Various doc fixes

### PR DESCRIPTION
**Pull request by ibgreen**

### CODE REVIEW

File: docs/upgrade-guide.md [hunk 1]
```diff
@@ -65,8 +66,8 @@ loggers for all AttributeManagers (i.e. for all layers).
 
 #### Removed method: `AttributeManager.addDynamic`
 
-This method has been deprecated since version 2.5 and is now removed, use
-`AttributeManager.add()` instead
+This method has been deprecated since version 2.5 and is now removed in v4.
+Use `AttributeManager.add()` instead.
```
Comment (shaojingli) [inline docs/upgrade-guide.md]: What about addInstanced?

File: docs/using-with-mapbox-gl.md [hunk 2]
```diff
@@ -7,11 +7,29 @@ the [react-map-gl](https://uber.github.io/react-map-gl/#/) component,
...
 As shown in the examples, the `DeckGL` React component works especially
 well as the child of a React component such as react-map-gl that displays
 a map using parameters similar to the deck.gl Viewport (i.e. latitude,
 longitude, zoom etc). In this configuration your deck.gl layers will
 render as a perfectly synchronized geospatial overlay over the underlying map.
+
+## About Mapbox Tokens
+
+To show maps from a service such as Mapbox you will need to register with
+Mapbox and get a "token" that you need to provide to your map component.
```
Comment (shaojingli) [inline docs/using-with-mapbox-gl.md]: register with Mapbox to get a "token" needed to display any map content from them.

File: docs/using-with-mapbox-gl.md [hunk 3]
```diff
@@ -7,11 +7,29 @@ the [react-map-gl](https://uber.github.io/react-map-gl/#/) component,
...
+with the map data provider (mapbox), and apply for a token.
+The token will usually be free until a certain level of traffic is exceeded.
+
+While the token will need to be hard-coded into your application in
+production, there are several ways to provide a token during development:
+* Modify file to specify your Mapbox token,
+* Set an environment variable (MapboxAccessToken) - through the use of a
+  webpack loader or browserify transform, see the hello-world examples
+  for details.
+* Provide a token in the URL. See documentation of react-map-gl.
```
Comment (ericsoco) [inline docs/using-with-mapbox-gl.md]: Thank you! Very thorough.

File: examples/README.md [hunk 4]
```diff
@@ -1,31 +1,79 @@
 # deck.gl Examples
 
-## Overview
+All deck.gl examples are set up to be "stand-alone", which means that
+you can copy the example folder to your own environment, run `yarn` or `npm install`
+and `npm start` and withing minutes have a running, minimal app that you can
```
Comment (ericsoco) [inline examples/README.md]: within~~g~~

File: docs/upgrade-guide.md (hunk 1 above)
Comment (ibgreen) [inline docs/upgrade-guide.md]: Good catch.

It was used in so many places and also when I started replacing it the app lines became a lot longer, forcing ugly line breaks. So... I let it be for the moment.

We should probably do a search for DEPRECATED in the code base and see if there are more old v3 things we can cut.

File: examples/README.md [hunk 5]
```diff
@@ -1,31 +1,79 @@
...
+* [line](./line/README.md)
+* [plot](./plot/README.md)
+* [scatterplot](./scatterplot/README.md)
+* [screen-grid](./screen-grid/README.md)
+* [trips](./trips/README.md)
+
+Remarks:
+* Since these examples are set up so they can be run both stand-alone and
+  also loaded by the website (demo), their code is organize slightly
+  differently than a completely stand-alone applications typicall are.
```
Comment (ericsoco) [inline examples/README.md]: typicall*y*

**Comment by ericsoco** [comment]
FYI I tested `npm run build` and `npm start` on Node v6.4 (per my doc update ^^) and I see no problems.

**Pull request merged**